### PR TITLE
feat: Helm chart updates for minnesota

### PR DESCRIPTION
**Pull request by bnevis-i**
List of changes:

- Brought README up-to-date
- Added support for authenticated pulls (to dockerhub)
- Added core-common-config microservice
- Brought command lines and environment variables up-to-date with EdgeX 3.0 docker-compose changes
- Added missing data volumes
- Created YAML values files to specify tip and releasee versions of containers
- Fixed bug in eKuiper volume initialization
- Repairs to ingress rules for port matching
- Documentation to obtain authentication JWT

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- [ ] I have fully tested (add details below) this the new feature or bug fix (if not, why?)

## Testing Instructions
Command to launch everything in distributed mode with security enabled:

```
helm install edgex-minnesota  -n edgex --values development-images-values-x86_64.yaml --set imagePullSecrets="{ dockerhub }" --set edgex.features.messagebusbackend=mqtt --set edgex.security.enabled=true --set edgex.storage.useHostPath=false --set edgex.storage.nonSharedVolumesClassName=ceph-block --set edgex.storage.sharedVolumesClassName=ceph-filesystem  --values runall.yaml . 
```

Where the edgex namespace already exists and the "dockerhub" image pull secret is generated according to the README.md.
If only have a single node storage cluster omit useHostPath=false and don't override the storageclass names.

The contents of runall.yaml is:
```yaml
edgex:

  # replicas defines the number of replicas in a Deployment for the respective application
  replicas:
    core:
      common_config: 1
      command: 1
      data: 1
      metadata: 1
    support:
      notifications: 1
      scheduler: 1
    appservice:
      rules: 1
      mqttexport: 1
      httpexport: 1
      appsample: 1
      metricsinfluxdb: 1
      rfidllrpinventory: 1
      externalmqtttrigger: 1
    device:
      virtual: 1
      rest: 1
      usbcamera: 1
      onvifcamera: 1
      mqtt: 1
      modbus: 1
      bacnet: 1
      coap: 1
      snmp: 1
      rfidllrp: 1
      gpio: 1
    ui: 1
    system: 1
    consul: 1
    redis: 1
    ekuiper: 1
    modbussimulator: 1
    mqttbroker: 1
    natsserver: 1
```

Expect that edgex-device-bacnet and edgex-device-coap will fail to launch.

### CODE REVIEW

File: deployment/helm/README.md [hunk 1]
```diff
@@ -1,16 +1,44 @@
...
 
 ## Prerequisites
 
-- Kubernetes cluster 1.10+
+- Kubernetes cluster 1.24+
 - [Helm](https://helm.sh/) 3.7.0+
 
-## Installation
+Before starting, make sure you have curl and openssl installed locally.
+These tools are need to generate TLS assets and test the configuration.
```
Comment (jim-wang-intel) [inline deployment/helm/README.md:17]:
```suggestion
These tools are needed to generate TLS assets and test the configuration.
```

File: deployment/helm/README.md [hunk 2]
```diff
@@ -1,16 +1,44 @@
...
+follow the instructions at
+<https://kubernetes.github.io/ingress-nginx/deploy/#quick-start>
+
+To install MetalLB,
+follow the instructions at
+<https://metallb.universe.tf/installation/#installation-with-helm>
+
+The example also arbitrarily configures the ingress route to
+respond to the hostname `edgex` 
+(which should be passed using the TLS Server Name Identification (SNI) feature).
```
Comment (jim-wang-intel) [inline deployment/helm/README.md:35]:
```suggestion
(which should be passed in using the TLS Server Name Identification (SNI) feature).
```

File: deployment/helm/README.md [hunk 3]
```diff
@@ -189,6 +214,27 @@ curl -iv --resolve edgex:443:<MetalLB IP> --cacert server-ca.pem --cert client.p
 {"apiVersion":"v2","timestamp":"Wed Feb  2 18:32:57 UTC 2022"}
 ```
 
+*** 
+TODO get an JWT from Helm
+***
```
Comment (jim-wang-intel) [inline deployment/helm/README.md:219]: details to be filled in?

Comment (bnevis-i) [reply deployment/helm/README.md]: This was a big hole.  I had to make a lot of changes.  No wonder I hadn't submitted PR review earlier.  This has been updated.

**Review by jim-wang-intel — APPROVED** [review]
LGTM

**Pull request merged**